### ISSUE TITLE
[Feature Request] Switch between VLC and browser

### ISSUE THREAD

**Issue by dispatchrabbi** [m1k1o/neko#71]
Heyo! First, let me say thank you for all the hard work you've put into this. I have a *ton* of appreciation for someone willing to fork a project, improve it, update it, and keep it active. I was able to get neko up and running yesterday with only a couple missteps or dead ends, and it's exactly as advertised. So again, thank you.

I am wondering if there's any way to set up neko so that you could switch between it showing a browser and VLC. I've got a group where it would be advantageous to be able to share both local content and online content, and right now - unless I've missed something - I'd need to have two instances of neko to support that. It'd be great to be able to do it without having to jump instances.

I understand that this is probably a fairly hefty feature request, but it would really go a long way toward making neko a true kosmi/kast/rabb.it replacement.

**Comment by mbattista** [comment]
My first suggestion would be to checkout the xfce Container and install firefox and VLC?

**Comment by m1k1o** [comment]
Yes, that is a good suggestion. With xfce you can get desktop-like experience and switching between multiple apps:

![image](https://user-images.githubusercontent.com/7534274/121814761-0ad57a00-cc73-11eb-9e72-269d5f7503ef.png)

**Comment by dispatchrabbi** [comment]
Thanks for the suggestion! I totally forgot there was an xfce build. I'll give it a shot and see how it goes.

I do still think it would be nice to be able to switch back and forth between (what seem to me to be) the two common use cases without having to deal with a whole desktop, but I can also see how you'd have higher priorities than trying to make that happen.

Thank you for your quick response!

**Comment by m1k1o** [comment]
That would be a definitly nice usecase, what you suggested. Let's keep this issue open to track progress on it.

**Comment by blitzmikyu** [comment]
> Yes, that is a good suggestion. With xfce you can get desktop-like experience and switching between multiple apps:
> 
> ![image](https://user-images.githubusercontent.com/7534274/121814761-0ad57a00-cc73-11eb-9e72-269d5f7503ef.png)

Not entirely related to the issue but with XFCE branch, is there a way to retain installed applications/settings? I need to reinstall vlc/firefox every time I restart or update the container.

**Comment by m1k1o** [comment]
You can mount any volume to that image and keep it persistent. E.G. mounting `/home/neko` folder for data and `/opt` where
 you install programs.

Although, installing programs with `apt install` can modify/create various files in the file system (in `/lib` or `usr`), you can either mount all of those or just simply install the programs in Dockerfile:

```Dockerfile
FROM m1k1o/neko:xfce

RUN set -eux; apt-get update; \
    apt-get install -y --no-install-recommends <your-dependencies>; \
    # clean up
    apt-get clean -y; \
    rm -rf /var/lib/apt/lists/* /var/cache/apt/*
```

And then just build it, and run with mounted `/home/neko` folder.

Alternatively, you can commit container before stopping container and save that.
